wordpress-mobile/WordPress-Android
# [Blogging Prompts v1] Implement dismiss action for local notifications

Fixes #16295 

To test:
1 - Onboarding notification
- Enable the blogging prompts notification feature flag (see `BloggingPromptsNotificationConfig`)
- Open app logged in and with a site selected
- Click on the blogging prompts onboarding notification OR in the "Learn more" button
- Blogging prompts onboarding dialog should be seen
- Tapping on "Learn more", "Dismiss" or anywhere else in the notification should dismiss it.

2 - Answer prompt notification
- Open `Debug settings` and enable `blogging_prompts_notification_remote_field`
- On `My Site`, tap on `Menu`
- Tap on `Site Settings`
- Tap on `Reminders and prompts` and select a date and time for the reminder
- Change the device date to something after the selected date/time for the reminder
- A notification to answer the prompt should appear
- Tap on the notification and the editor should be opened
- Tapping on "Answer", "Dismiss" or anywhere else in the notification should dismiss it.
- If the feature flag mentioned above is disabled, the existing blogging reminder notification should be shown 

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing for both notifications

3. What automated tests I added (or what prevented me from doing so)
Updated `BloggingPromptsOnboardingNotificationSchedulerTest.kt` unit test.

PR submission checklist:
